src/components/Page/components/Header/Header.tsx

```diff
@@ -91,13 +74,16 @@ class Header extends React.PureComponent<CombinedProps, State> {
       console.warn(intl.translate('Polaris.Page.Header.iconWarningMessage'));
     }
 
+    const hasRollupActions =
+      secondaryActions.length > 0 || actionGroups.length > 0;
+
     const className = classNames(
       styles.Header,
       titleHidden && styles['Title-hidden'],
-      pagination && styles['Header-hasPagination'],
-      separator && styles['Header-hasSeparator'],
       breadcrumbs && breadcrumbs.length && styles['Header-hasBreadcrumbs'],
-      this.hasRollup && styles['Header-hasRollup'],
+      separator && styles['Header-hasSeparator'],
```

Comment:
Is this really the preferred style for variants? I thought we just did: `separator && styles.separator`

Reply:
Seems off, I wouldn't follow that convention.